src/kboot.c

```diff
@@ -729,6 +729,57 @@ static int dt_set_ipd(void)
     return 0;
 }
 
+#define NVRAM_START    0x700000
+#define NVRAM_MAX_SIZE SZ_1M
+
+static int dt_set_nvram(void)
+{
+    int adt_path[8];
+    u64 start, size;
+
+    int anode = adt_path_offset_trace(adt, "/arm-io/spi1/spinor/anvram", adt_path);
+    if (anode < 0) {
+        printf("ADT: nvram partition not found\n");
+        return 0;
+    }
+
+    int pp = 0;
+    while (adt_path[pp])
+        pp++;
+    adt_path[pp + 1] = 0;
+
+    int ret = adt_get_reg(adt, adt_path, "reg", 0, &start, &size);
+    if (ret < 0) {
+        printf("ADT: could not read nvram partition start/size\n");
+        return 0;
+    }
+
+    if (start != NVRAM_START || size > NVRAM_MAX_SIZE) {
+
+        printf("ADT: unexpected nvram partition start (0x%lx) size (0x%lx)\n", start, size);
+        return 0;
+    }
+
+    int node = fdt_path_offset(dt, "nvram");
+    if (node < 0) {
+        printf("DT: nvram alias/partition not found\n");
+        return 0;
+    }
+
+    fdt32_t reg[2];
+    fdt32_st(reg + 0, start);
+    fdt32_st(reg + 1, start);
```

Comment:
I guess this should set `size`. Otherwise the value is unused.
```suggestion
    fdt32_st(reg + 1, size);
```

Reply:
thanks for spotting this, fixed. Strange that my dump from j493 using 13.5 seemingly reported the expected mtd0 size.

Reply:
Hmm... and what does `mtdinfo /dev/mtd0` say on your machine?

Reply:
Btw, I still don't know how the "current bank" is picked. I thought each nvram change would increase the counter but now I tried again and the value is still 3 which would suggest the changes are made in place.

Maybe in your case the current bank is within those first 128 kB, so you could see the changes.